zinc/src/sbt-test/source-dependencies/java-class-invs/test

```diff
@@ -0,0 +1,6 @@
+> app/compile
+$ sleep 1000
+$ copy-file changes/Std.scala std/Std.scala
+$ delete lib/target/early/output.jar
+# delete early/output.jar to avoid it accidentally working by merging into an older jar
```

Comment:
So what this is saying is that the incremental compilation would work if `early/output.jar` is not magically deleted? If so we should account for that particular scenario instead of invalidating all Java sources in all cycles.

Reply:
Good point.

The difference between this working here and not working elsewhere is because zinc-scripted adds target/classes to the classpath.

There are suspicions that might not work well in pruning classfiles on java source deletions, but I need to write the test and see.  Drafting this in the meantime.